frontend/src/pages/verify.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect } from "react";
+import { useRouter } from "next/router";
+import { auth } from "@/utils/firebase";
+import WelcomeTemplate from "@/components/templates/WelcomeTemplate";
+import Button from "@/components/atoms/Button";
+import CircularProgress from "@mui/material/CircularProgress";
+import {
+  useSignInWithEmailAndPassword,
+  useSendEmailVerification,
+} from "react-firebase-hooks/auth";
+import { useQueryClient } from "@tanstack/react-query";
+import { useAuth } from "@/utils/AuthContext";
+import { fetchUserIdFromDatabase, updateVerifiedStatus } from "@/utils/helpers";
+
+const Verify = () => {
+  const router = useRouter();
+  const [sendEmailVerification, sending, emailError] =
+    useSendEmailVerification(auth);
+
+  const { user, loading, error, signOutUser } = useAuth();
+  const queryClient = useQueryClient();
+
+  // useEffect(() => {
+  //   const checkUserVerification = async () => {
+  //     try {
+  //       // Check if user is signed in
+  //       const user = auth.currentUser;
+  //       const emailParam = user?.email as string;
+  //       console.log(user);
+
+  //       // Fetch user ID asynchronously
+  //       const userId = await fetchUserIdFromDatabase(emailParam);
+  //       console.log(userId);
+  //       if (user) {
+  //         // Check if user's email is verified
+  //         if (user.emailVerified) {
+  //           // If email is verified, redirect to home page
+  //           updateVerifiedStatus(userId, true);
+  //           router.push("/events/view");
+  //         } else {
+  //           // If email is not verified, stay on verification page
+  //           console.log("Email not verified");
+  //         }
+  //       } else {
+  //         // If user is not signed in, redirect to login page
+  //         router.push("/login");
+  //       }
+  //     } catch (error) {
+  //       console.error("Error checking user verification:", error);
+  //     }
+  //   };
+
+  //   checkUserVerification();
+  // }, []);
```

Comment:
I have commented out this code because I pushed some new code to AuthContext.tsx. The code I pushed will check if a user is verified or not and route them accordingly. This means there will be no need to do that check here again. 

Your new changes will simply be functions that will be called when an event happens. For now, let us leave firebase as the source of truth for email verification flag